siaskynet/_upload.py

```diff
@@ -18,6 +18,67 @@ def default_upload_options():
     return obj
 
 
+def upload(self, upload_data, custom_opts=None):
+    """Uploads the given generic data and returns the skylink."""
+
+    response = self.upload_request(upload_data, custom_opts)
+    sia_url = utils.uri_skynet_prefix() + response.json()["skylink"]
+    response.close()
+    return sia_url
+
+
+def upload_request(self, upload_data, custom_opts=None):
+    """Uploads the given generic data and returns the response object."""
+
+    opts = default_upload_options()
+    opts.update(self.custom_opts)
+    if custom_opts is not None:
+        opts.update(custom_opts)
+
+    if len(upload_data) == 1:
+        fieldname = opts['portal_file_fieldname']
+        # this appears to be missing in go sdk; maybe the server ignores it?
```

Comment:
Please write an issue on the Go SDK instead of a comment here, as this will go forgotten.

Reply:
I've asked about this at https://github.com/NebulousLabs/go-skynet/issues/26

Reply:
Awesome, thanks 🙂

Reply:
Can you also remove this comment?

Reply:
yes